app/models/terminal_executor.rb

```diff
@@ -36,7 +36,11 @@ def execute!(*commands)
   def stop!(signal)
     # Need pkill because we want all
     # children of the parent process dead
-    system "pkill", "-#{signal}", "-P", pid.to_s if pid
+    if pgid
+      system('kill', "-#{signal}", "-#{pgid}")
+    elsif pid
+      system "pkill", "-#{signal}", "-P", pid.to_s
+    end
```

Comment:
Consistent use of parens, please, for those of us with mild OCD tendencies.  :-)